src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -22,22 +21,41 @@
     BindingConstraintFrequency,
     BindingConstraintOperator,
     BindingConstraintProperties,
+    BindingConstraintPropertiesLocal,
+    ClusterData,
     ConstraintMatrixName,
     ConstraintTerm,
+    LinkData,
 )
 from antares.service.base_services import BaseBindingConstraintService
 from antares.tools.ini_tool import IniFile, IniFileTypes
 from antares.tools.matrix_tool import df_save
 from antares.tools.time_series_tool import TimeSeriesFileType
 
 
+def serialize_term_data(
```

Comment:
nice to add this utility method but I believe you should just have the term as an argument and treat offset and wieght internally as they are class variables.

In fact I think this should be a method of the class ConstraintTerm. It would be even better

Also i think we have the same writing for both terms so I don't understand the if else return None